hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -1078,4 +1088,138 @@ public void close() {
     this.heartbeatClient.stop();
     this.txnManager.close();
   }
+
+  /**
+   * add columns to table.
+   *
+   * @param colName col name to be added. if we want to add col to a nested filed, the fullName should be specify
+   * @param schema col type to be added.
+   * @param doc col doc to be added.
+   * @param position col position to be added
+   * @param positionType col position change type. now support three change types: first/after/before
+   */
+  public void addCol(String colName, Schema schema, String doc, String position, TableChange.ColumnPositionChange.ColumnPositionType positionType) {
+    Pair<InternalSchema, HoodieTableMetaClient> pair = getInternalSchemaAndMetaClient();
+    InternalSchema newSchema = SchemaChangePersistHelper
+        .applyAddChange(pair.getLeft(), colName, AvroInternalSchemaConverter.convertToField(schema), doc, position, positionType);
+    commitTableChange(newSchema, pair.getRight());
+  }
+
+  public void addCol(String colName, Schema schema) {
+    addCol(colName, schema, null, null, null);
```

Comment:
This is because such operations as hive / spark / MySQL use string as a parameter。 eg: alter table xxx add columns(name string **after**/**before** id .     i think it will be better to use  after/before/first as param

Reply:
using enum value "NO_OPERATION" instead of use "NULL_COLUMN" in ColumnPositionType